distributed/scheduler.py

```diff
@@ -7314,7 +7506,39 @@ def request_remove_replicas(
         )
 
 
-def _remove_from_processing(state: SchedulerState, ts: TaskState) -> str | None:
+def _add_to_processing(
+    state: SchedulerState, ts: TaskState, recommendations: dict, worker_msgs: dict
+) -> WorkerState | None:
+    if state.validate:
+        assert not ts.waiting_on
+        assert not ts.who_has
+        assert not ts.exception_blame
+        assert not ts.processing_on
+        assert not ts.has_lost_dependencies
+        assert ts not in state.unrunnable
+        assert all(dts.who_has for dts in ts.dependencies)
+
+    if ws := state.decide_worker(ts, recommendations):
+        state._set_duration_estimate(ts, ws)
+        ts.processing_on = ws
+        ts.state = "processing"
+        state.queued.discard(ts)
+        state.acquire_resources(ts, ws)
+        state.check_idle_saturated(ws)
+        state.n_tasks += 1
+
+        if ts.actor:
+            ws.actors.add(ts)
+
+        # logger.debug("Send job to worker: %s, %s", worker, key)
+
+        worker_msgs[ws.address] = [_task_to_msg(state, ts)]
+    return ws
+
+
+def _remove_from_processing(
```

Comment:
I guess the point being, I never thought of this or `_add_to_memory` as meaning "this adds the task to memory". You wouldn't make a function just to do `ts.state = "memory"`. Rather, I thought of it as being "this does all the things that need to happen when a task is going to be added to memory", since there are a few different transitions from which that can happen. Same now for `_remove_from_processing`. What would be a better name for that?